engine/src/state_chain_observer/client.rs

```diff
@@ -795,52 +806,84 @@ impl SignedExtrinsicClientBuilderTrait for SignedExtrinsicClientBuilder {
 					recorded_version, *CFE_VERSION
 				);
 
-				let block_hash = finalized_block_stream.cache().hash;
-				let block_number = finalized_block_stream.cache().number;
-
-				// Submitting transaction with subxt sometimes gets stuck without returning any
-				// error (see https://linear.app/chainflip/issue/PRO-1064/new-cfe-version-gets-stuck-on-startup),
-				// so we use a timeout to ensure we can recover:
-				tokio::time::timeout(CFE_VERSION_SUBMIT_TIMEOUT, async {
-					let current_nonce = rpc_client
-						.request::<u32>(
-							"system_accountNextIndex",
-							subxt::rpc_params![&subxt_signer.account_id()],
-						)
-						.await?;
+				// Submitting the CFE version has three possible results:
+				//   - Successful
+				//   - Returns no error, but not successful
+				//   - Returns error
+				// Here we call the submission code multiple times before giving up with an error.
+				const MAX_UPDATE_VERSION_RETRIES: usize = 10;
+				let mut update_successful = false;
+				for retry in 0..MAX_UPDATE_VERSION_RETRIES {
```

Comment:
We can avoid the retry+1s below :)
```suggestion
				for retry in 1..=MAX_UPDATE_VERSION_RETRIES {
```